Examples/Scripts/FreeSurferPipelineBatch-long.sh

```diff
@@ -0,0 +1,103 @@
+Template_list="HCA6002236_V1_V2"
+
+EnvironmentScript="/media/myelin/brainmappers/Connectome_Project/MishaLongitudinal/hcp/scripts/SetUpHCPPipeline-long.sh" #Pipeline environment script
+
+# Requirements for this script
+#  installed versions of: FSL, FreeSurfer, Connectome Workbench (wb_command), gradunwarp (HCP version)
+#  environment: HCPPIPEDIR, FSLDIR, FREESURFER_HOME, CARET7DIR, PATH for gradient_unwarp.py
+
+# If you want to use FreeSurfer 5.3, change the ${queuing_command} line below to use
+# ${HCPPIPEDIR}/FreeSurfer/FreeSurferPipeline-v5.3.0-HCP.sh
+
+#Set up pipeline environment variables and software
+source "$EnvironmentScript"
+
+# Log the originating call
+echo "$@"
+
+#NOTE: syntax for QUEUE has changed compared to earlier pipeline releases,
+#DO NOT include "-q " at the beginning
+#default to no queue, implying run local
+QUEUE=""
+#QUEUE="hcp_priority.q"
+
+########################################## INPUTS ########################################## 
+
+#Scripts called by this script do assume they run on the outputs of the PreFreeSurfer Pipeline
+
+######################################### DO WORK ##########################################
+
+Template_list=( ${Template_list[@]} )
+Timepoint_list=( ${Timepoint_list[@]} )
+Subjlist=( ${Subjlist[@]} )
+
+for i in ${!Subjlist[@]}; do
+  Subject=${Subjlist[i]}
+  #Subject's time point list, @ separated.  
+  TPlist="${Timepoint_list[i]}"
+  #Array with timepoints
+  IFS=@ read -ra Timepoints <<< "$TPlist"
+  #Freesurfer longitudinal average template label
+  LongitudinalTemplate=${Template_list[i]}
+
+  #Longitudinal FreeSurfer Input Variables
+  SubjectID="$Subject" #FreeSurfer Subject ID Name
+  SubjectDIR="${StudyFolder}/${Subject}/T1w" #Location to Put FreeSurfer Subject's Folder
+  T1wImage="${StudyFolder}/${Subject}/T1w/T1w_acpc_dc_restore.nii.gz" #T1w FreeSurfer Input (Full Resolution)
+  T1wImageBrain="${StudyFolder}/${Subject}/T1w/T1w_acpc_dc_restore_brain.nii.gz" #T1w FreeSurfer Input (Full Resolution)
+  T2wImage="${StudyFolder}/${Subject}/T1w/T2w_acpc_dc_restore.nii.gz" #T2w FreeSurfer Input (Full Resolution)
+
+  if [[ "${command_line_specified_run_local}" == "TRUE" || "$QUEUE" == "" ]] ; then
+      echo "About to locally run ${HCPPIPEDIR}/FreeSurfer/LongitudinalFreeSurferPipeline.sh"
+      #NOTE: fsl_sub without -q runs locally and captures output in files
+      queuing_command=("$FSLDIR/bin/fsl_sub")
+  else
+      echo "About to use fsl_sub to queue ${HCPPIPEDIR}/FreeSurfer/LongitudinalFreeSurferPipeline.sh"
+      queuing_command=("$FSLDIR/bin/fsl_sub" -q "$QUEUE")
+  fi
+
+  #DO NOT PUT timepoint-specific options here!!!
+  echo "${queuing_command[@]}" "$HCPPIPEDIR"/FreeSurfer/LongitudinalFreeSurferPipeline.sh --subject="$Subject" --path="$StudyFolder" --sessions \
+    "$TPlist" --template-id "$LongitudinalTemplate" --extra-reconall-arg-long=-T2pial \
+    --extra-reconall-arg-base=-T2pial --extra-reconall-arg-base=-T2 \
```

Comment:
I don't think we want to need to put these extra recon all arguments into the batch.  We should define the correct way to run the pipeline in the pipeline itself.

Reply:
Agree.  If there are a set of defaults that should be used, those should be the defaults defined within the pipeline script itself.

Reply:
I removed extra arguments option from the LongitudinalFreeSurferPipeline.sh, updating its batch accordingly. Now the only option is "use-T2w" which triggers a certain set of extra options for template and timepoints in the pipeline script.